modin/pandas/series.py

```diff
@@ -1543,16 +1594,20 @@ def repeat(self, repeats, axis=None):  # noqa: PR01, RT01, D200
         return self.__constructor__(query_compiler=self._query_compiler.repeat(repeats))
 
     def reset_index(
-        self, level=None, drop=False, name=None, inplace=False
+        self, level=None, drop=False, name=no_default, inplace=False
     ):  # noqa: PR01, RT01, D200
         """
         Generate a new Series with the index reset.
         """
+        if name is no_default:
+            # For backwards compatibility, keep columns as [0] instead of
+            #  [None] when self.name is None
+            name = 0 if self.name is None else self.name
+
         if drop and level is None:
             new_idx = pandas.RangeIndex(len(self.index))
             if inplace:
                 self.index = new_idx
-                self.name = name or self.name
```

Comment:
Why is this different?

Reply:
pandas behavior changed, so we set the name below.